packages/devtools-network-console/src/host/vscode-protocol-host.ts

```diff
@@ -352,6 +338,34 @@ export default class VsCodeProtocolHost implements INetConsoleHost {
             ...message,
         });
     }
+
+    public reassociateFocus() {
```

Comment:
We actually do "push the last focusable element" onto a focus-restore stack, and this is only a fallback. 

The problem that this particular change solves is that the Host can invoke focus into the Frontend, for example, when doing a right-click Edit Authorization operation on a collection (or editing an Environment's variables). Because the modal in the Frontend asserts focus, it tries first to determine the document's active element. However, when this is coming from the Host, in general, the Frontend doesn't have DOM access due to cross-origin DOM access prohibition.

I considered sending a message from the Frontend to the Host instructing it to reassociate focus. However, it isn't clear that this is really a good play. 

When a modal is invoked now, it attempts to determine the last-focused element. If it's within the Frontend DOM, it is stored in a stack to restore. If it isn't, then it assumes focus belongs to the Host, but since we can't reasonably be sure what element in the Host should restore focus, that's when this particular function is called.

https://github.com/microsoft/edge-devtools-network-console/pull/43/files/d7546e6652f3bafd2d6bda88c2222ec534cd71fa?file-filters%5B%5D=.html&file-filters%5B%5D=.md&file-filters%5B%5D=.ts#diff-f1b366283bcf526899aef23e389484827912090c028d4f14361aecfaafae82f0

Reply:
Ahh, that makes sense. Thanks for explaining!